### DISCUSSION TITLE
UNSOLVABLE: Property `interactive: true` results in wrong prompt length calculation in Bash/Zsh

### DISCUSSION THREAD

**Discussion by lewis-yeung** [JanDeDobbeleer/oh-my-posh#5455] [General]
In Bash/Zsh, when the property `interactive` is `true` for a segment, the prompt length calculation can be wrong because of possible string expansions (e.g., `\w` in Bash and `%d` in Zsh which both expand to the current working directory), thus a right-aligned block is not being properly right-aligned.

Unfortunately, it's not possible for OMP to know the final prompt length since the string expansion is done by shells.

**Comment by lewis-yeung** [comment]
I'm posting this just in case someone is confused by it.

**Reply by JanDeDobbeleer** [reply]
True. This is "at your own risk".
